binaries/speed_benchmark_torch.cc

```diff
@@ -200,7 +204,10 @@ int main(int argc, char** argv) {
     std::cout << module.forward(inputs) << std::endl;
   }
 
-  c10::WithCPUCachingAllocatorGuard cachine_allocator_guard;
+  std::unique_ptr<c10::CPUCachingAllocator> caching_allocator =
+    std::make_unique<c10::CPUCachingAllocator>();
+  c10::WithCPUCachingAllocatorGuard cachine_allocator_guard(
+      caching_allocator.get(), FLAGS_use_caching_allocator);
```

Comment:
How about

```
c10::optional<c10::WithCPUCachingAllocatorGuard> maybe_guard;
if (FLAGS_use_caching_allocator) {
  maybe_guard.emplace(caching_allocator);
}
```

?  This way you could remove a concept from the API.

Reply:
Sounds fine.